Add plugin: Draw Steel Elements

scottTomaszewski opened the pull request:
Add Draw Steel Elements Plugin

# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/scottTomaszewski/draw-steel-elements

## Release Checklist
- [x] I have tested the plugin on
  - [ ]  Windows
  - [ ]  macOS
  - [x]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [x] My GitHub release contains all required files (as individual files, not just in the source.zip / source.tar.gz)
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] My README.md describes the plugin's purpose and provides clear usage instructions.
- [x] I have read the developer policies at https://docs.obsidian.md/Developer+policies, and have assessed my plugins's adherence to these policies.
- [x] I have read the tips in https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[export function drawSteelAdmonitionPlugin(admonitions: DrawSteelAdmonition[]) {](https://github.com/scottTomaszewski/draw-steel-elements/blob/ddd0dae9ea90ad881dcc6d979024ac7f53780724/src/drawSteelAdmonition/drawSteelAdmonitionExtension.ts#L9-L48), [export class DrawSteelAdmonitionsPostProcessor {](https://github.com/scottTomaszewski/draw-steel-elements/blob/ddd0dae9ea90ad881dcc6d979024ac7f53780724/src/drawSteelAdmonition/drawSteelAdmonitionPostProcessor.ts#L5-L18)
Shouldn't this be a [MarkdownCodeBlockPostProccessor](https://docs.obsidian.md/Plugins/Editor/Markdown+post+processing#Post-process+Markdown+code+blocks) ?

[const yaml = YAML.parse(codeElement.getText());](https://github.com/scottTomaszewski/draw-steel-elements/blob/ddd0dae9ea90ad881dcc6d979024ac7f53780724/src/drawSteelAdmonition/powerRollAdmonition.ts#L29)
Use the `parseYaml` function from the Obsidian API instead of importing a seperate library.

[this.addSettingTab(new DrawSteelAdmonitionSettingTab(this.app, this));](https://github.com/scottTomaszewski/draw-steel-elements/blob/ddd0dae9ea90ad881dcc6d979024ac7f53780724/main.ts#L20)
Your settings tab is currently empty, it should not be shown in that case.

[view?.previewMode.rerender(true);](https://github.com/scottTomaszewski/draw-steel-elements/blob/ddd0dae9ea90ad881dcc6d979024ac7f53780724/main.ts#L43)
Use [`this.workspace.updateOptions()`](https://docs.obsidian.md/Reference/TypeScript+API/Workspace/updateOptions) instead.

scottTomaszewski commented:
> [export function drawSteelAdmonitionPlugin(admonitions: DrawSteelAdmonition[]) {](https://github.com/scottTomaszewski/draw-steel-elements/blob/ddd0dae9ea90ad881dcc6d979024ac7f53780724/src/drawSteelAdmonition/drawSteelAdmonitionExtension.ts#L9-L48), [export class DrawSteelAdmonitionsPostProcessor {](https://github.com/scottTomaszewski/draw-steel-elements/blob/ddd0dae9ea90ad881dcc6d979024ac7f53780724/src/drawSteelAdmonition/drawSteelAdmonitionPostProcessor.ts#L5-L18) Shouldn't this be a [MarkdownCodeBlockPostProccessor](https://docs.obsidian.md/Plugins/Editor/Markdown+post+processing#Post-process+Markdown+code+blocks) ?
> 
> [const yaml = YAML.parse(codeElement.getText());](https://github.com/scottTomaszewski/draw-steel-elements/blob/ddd0dae9ea90ad881dcc6d979024ac7f53780724/src/drawSteelAdmonition/powerRollAdmonition.ts#L29) Use the `parseYaml` function from the Obsidian API instead of importing a seperate library.
> 
> [this.addSettingTab(new DrawSteelAdmonitionSettingTab(this.app, this));](https://github.com/scottTomaszewski/draw-steel-elements/blob/ddd0dae9ea90ad881dcc6d979024ac7f53780724/main.ts#L20) Your settings tab is currently empty, it should not be shown in that case.
> 
> [view?.previewMode.rerender(true);](https://github.com/scottTomaszewski/draw-steel-elements/blob/ddd0dae9ea90ad881dcc6d979024ac7f53780724/main.ts#L43) Use [`this.workspace.updateOptions()`](https://docs.obsidian.md/Reference/TypeScript+API/Workspace/updateOptions) instead.

> Shouldn't this be a MarkdownCodeBlockPostProccessor?

Yeah, sorry I pulled code from another plugin of mine that was dynamically building processors.  Ive updated the code to simplify it down to only what is needed

> Use the parseYaml function

Did not know this existed, thank you! Updated.

> Your settings tab is currently empty

Settings stuff removed

> Use this.workspace.updateOptions() instead.

Removed re-render since logic wasnt used

The `main` branch has been updated and the `0.0.6` release made to reflect changes

scottTomaszewski commented:
Sorry, I know this is abnormal, but I moved (transferred) the repo under a new organization, but its the same (updated) code.  Ive updated the PR to reflect the transfer.  Let me know if you need me to open a different PR or anything, and thanks for being accommodating!
